doc/source/whatsnew/v2.0.0.rst

```diff
@@ -376,6 +376,7 @@ Missing
 - Bug in :meth:`Index.equals` raising ``TypeError`` when :class:`Index` consists of tuples that contain ``NA`` (:issue:`48446`)
 - Bug in :meth:`Series.map` caused incorrect result when data has NaNs and defaultdict mapping was used (:issue:`48813`)
 - Bug in :class:`NA` raising a ``TypeError`` instead of return :class:`NA` when performing a binary operation with a ``bytes`` object (:issue:`49108`)
+- Bug in :meth:`Dataframe.update` raising ``TypeError`` when column has NaT values (:issue:`16713`)
```

Comment:
Could we clarify please - if I understand correctly, it was only raising a TypeError if the column had NaT values and it wasn't present in the other DataFrame and you were using `overwrite=False`

Also, if you write `NaT`, make sure to put double ticks around it, like this
```
``NaT``
```

Reply:
Updated the message, let me know if the new one works!